### PR TITLE
DDF-4424 Updating the CDM (in-place) to update the observer state only on a successful ingest to the catalog framework

### PR DESCRIPTION
**Pull request by Kjames5269**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: #4223 
____

@vinamartin @lessarderic @clockard @kcover @peterhuffer 

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Kjames5269** [comment]
build now

File: catalog/catalog-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -123,6 +123,7 @@
              description="Monitors directories to process content files.">
         <feature>apache-commons</feature>
         <bundle>mvn:ddf.catalog.core/catalog-core-camelcontext/${project.version}</bundle>
+        <bundle>mvn:com.google.code.gson/gson/${gson.version}</bundle>
```
Comment (Kjames5269) [inline catalog/catalog-app/src/main/resources/features.xml]: ❓ This isn't being picked up from the CDM specific Itests. Anyone have ideas why?

Comment (peterhuffer) [reply catalog/catalog-app/src/main/resources/features.xml]: ContentDirectoryMonitorIT? That test does not start the feature, it starts the necessary bundles individually. 
https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java#L308-L344

Comment (Kjames5269) [reply catalog/catalog-app/src/main/resources/features.xml]: Thanks that was it 😄

**Comment by Kjames5269** [comment]
build now

**Comment by Kjames5269** [comment]
build now
(got unlucky with networks)

**Comment by jhunzik** [comment]
build now
I wasn't able to get that test to fail again locally

**Comment by Kjames5269** [comment]
build now

**Comment by jhunzik** [comment]
Hero Successful
The CDM was capable of handling a network disconnect by saving its state and resuming from where it left off after reconnecting.
Note:
During one of the testing attempts I observed some of the files not being properly ingested into the catalog. This was due to either the catalog being unavailable or the site names being incorrect. Camel would report back that the files were successfully queued for ingest but the catalog was never able to actually ingest them.

**Pull request merged**